### PR TITLE
Remove pricefeed config toml entirely.

### PR DESCRIPTION
**Pull request by clemire**
### Changelist
- Remove pricefeed config toml from protocol entirely:
  - toml generation functionality, including post-init hook to create file on protocol init
  - all code dependencies
- Update daemon to configure itself from the static exchange startup config in code.
- Rename `ExchangeStartupConfig` to `ExchangeQueryConfig`, which IMO is more specific and self-documenting.

As part of CORE-469, I plan to follow up with a PR that allows users to pass in an optional pricefeed daemon flag populated with json config that disables and/or changes query configuration of particular exchanges.

### Test Plan
test-ci and container-test pass locally.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/cmd/dydxprotocold/cmd/init.go
Comment (clemire) [inline protocol/cmd/dydxprotocold/cmd/init.go:1]: This file added a hook to the init command that generated pricefeed config toml.

File: protocol/daemons/configs/default_pricefeed_exchange_config.go
Comment (clemire) [inline protocol/daemons/configs/default_pricefeed_exchange_config.go:1]: This directory contained logic for generating and reading pricefeed config tomls.

File: protocol/daemons/pricefeed/client/client.go
Comment (clemire) [inline protocol/daemons/pricefeed/client/client.go:120]: type rename: `ExchangeStartupConfig` -> `ExchangeQueryConfig`

File: protocol/testing/containertest/config/pricefeed_exchange_config.toml
Comment (clemire) [inline protocol/testing/containertest/config/pricefeed_exchange_config.toml:1]: Remove pricefeed config from containertests.

File: protocol/testing/genesis.sh
Comment (clemire) [inline protocol/testing/genesis.sh:1479]: dydx token was not accounted for here when I added it a few weeks ago, this caused problems because it mentions non-existent markets as adjustment markets. Was not a problem before because we had only enabled the test exchange for container tests.

RN, container tests are running a daemon with all exchanges. I can update this once I add a flag that allows users to disable exchanges.

Comment (teddyding) [reply protocol/testing/genesis.sh:1479]: Could run `make update-sample-pregenesis` to check if prod_sample_genesis is changed as a result of this?

File: protocol/daemons/pricefeed/client/client.go
Comment (teddyding) [reply protocol/daemons/pricefeed/client/client.go:120]: Probably should rename variable `exchangeIdToStartupConfig` accordingly?

File: protocol/app/app.go
Comment (teddyding) [inline protocol/app/app.go:605]: For my understanding:
- This PR is no-op for the actual value of `exchangeStartupConfig` right? Meaning it's still initialized as the same value as before, just not read from a config file
- Is there currently any sanity check for the `exchangeStartupConfig` being populated correctly?

**Review by ttl33 — APPROVED** [review]
approved, pls address Teddy's comments and get his approval before merging!

**Review by BrendanChou — APPROVED** [review]

File: protocol/app/app.go
Comment (clemire) [reply protocol/app/app.go:605]: Yes and yes. See [this file](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/daemons/pricefeed/client/constants/static_exchange_startup_config_test.go).

File: protocol/testing/genesis.sh
Comment (clemire) [reply protocol/testing/genesis.sh:1479]: No updates!

File: protocol/daemons/pricefeed/client/client.go
Comment (clemire) [reply protocol/daemons/pricefeed/client/client.go]: Ok, renamed everywhere!

**Review by teddyding — APPROVED** [review]

**Pull request merged**
